install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl

```diff
@@ -19,30 +19,94 @@ spec:
           # line args instead of via config map. This allows users to inspect
           # the values used in past runs by inspecting the completed pod.
           args:
-            - "--cilium-namespace={{ .Release.Namespace }}"
             {{- if .Values.debug.enabled }}
             - "--debug"
             {{- end }}
             - "--ca-generate"
             - "--ca-reuse-secret"
-            {{- if and .Values.tls.ca.cert .Values.tls.ca.key }}
+            - "--ca-secret-namespace={{ .Release.Namespace }}"
             - "--ca-secret-name=cilium-ca"
-            {{- end }}
-            - "--hubble-server-cert-generate"
-            - "--hubble-server-cert-common-name={{ list "*" (.Values.cluster.name | replace "." "-") "hubble-grpc.cilium.io" | join "." }}"
-            - "--hubble-server-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- if .Values.hubble.relay.enabled }}
-            - "--hubble-relay-client-cert-generate"
-            - "--hubble-relay-client-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- end }}
-            {{- if and .Values.hubble.relay.enabled .Values.hubble.relay.tls.server.enabled }}
-            - "--hubble-relay-server-cert-generate"
-            - "--hubble-relay-server-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- end }}
-            {{- if and .Values.hubble.metrics.enabled .Values.hubble.metrics.tls.enabled }}
-            - "--hubble-metrics-server-cert-generate"
-            - "--hubble-metrics-server-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- end }}
+            - "--ca-common-name=Cilium CA"
+          env:
+            - name: CILIUM_CERTGEN_CONFIG
```

Comment:
That would definitely be a possibility (and also supported by certgen). It seemed just a bit overkill to me in this case due to the requirement for an additional resource, which can instead be avoided with this approach, while being functionally equivalent.

Reply:
Another advantage here is that the full pod manifest gets collected in sysdumps, while it would not be the case if provided through an external configmap.

Reply:
Fair, though we can collect the ConfigMap in the sysdump too. Not a blocker, just thought I'd check.

Reply:
Sure, it just requires less changes this way.